### PR TITLE
Add calico cni plugin

### PR DESCRIPTION
**Pull request by cmluciano**
This is a rebase on @klizhentas 's work as well as @ozdanborne 's trial commit.

I'm testing this manually at the moment.

Related kubernetes/kubeadm#218

### CODE REVIEW

**Comment by cmluciano** [comment]
Errors in deploy log

```
rom server (Invalid): error when creating ".tmp/calico-etcd-service.json": Service "calico-etcd" is invalid: spec.clusterIP: Invalid value: "10.96.232.136": provided IP is not in the valid range
                                                                                                                         [error validating ".tmp/calico-cni-plugin-clusterrole.json": error validating data: the server could not find the requested resource; if you choose to ignore these errors, turn validation off with --validate=false, error validating ".tmp/calico-cni-plugin-clusterrolebinding.json": error validating data: the server could not find the requested resource; if you choose to ignore these errors, turn validation off with --validate=false, error validating ".tmp/calico-etcd-daemonset.json": error validating data: found invalid field tolerations for v1.PodSpec; if you choose to ignore these errors, turn validation off with --validate=false, error validating ".tmp/calico-node-daemonset.json": error validating data: found invalid field tolerations for v1.PodSpec; if you choose to ignore these errors, turn validation off with --validate=false, error validating ".tmp/calico-policy-controller-clusterrole.json": error validating data: the server could not find the requested resource; if you choose to ignore these errors, turn validation off with --validate=false, error validating ".tmp/calico-policy-controller-clusterrolebinding.json": error validating data: the server could not find the requested resource; if you choose to ignore these errors, turn validation off with --validate=false, error validating ".tmp/calico-policy-controller-deployment.json": error validating data: found invalid field tolerations for v1.PodSpec; if you choose to ignore these errors, turn validation off with --validate=false]
                               Makefile:57: recipe for target 'addons' failed
                                                                             make: *** [addons] Error 1
```

Things I need to fix:
- [ ] Change range for CIDR
- [ ] validate referenced json

File: phase3/calico/calico-node-daemonset.json [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                                "name": "CALICO_DISABLE_FILE_LOGGING", 
+                                "value": "true"
+                            }, 
+                            {
+                                "name": "FELIX_DEFAULTENDPOINTTOHOSTACTION", 
+                                "value": "ACCEPT"
+                            }, 
+                            {
+                                "name": "CALICO_IPV4POOL_CIDR", 
+                                "value": "192.168.0.0/16"
```
Comment (ozdanborne) [inline phase3/calico/calico-node-daemonset.json]: I don't think this is the cause of the error, but this needs to be updated to match the cluster-cidr: 10.244.0.0/16

Comment (cmluciano) [inline phase3/calico/calico-node-daemonset.json]: 👍🏽

**Comment by klizhentas** [comment]
@cmluciano I suggest we merge one plugin e.g. flannel first to keep the PR footprint small, rather than merging it all at once, what do you think? I'm currently on vacation, but can help next week once I'm back.

**Comment by cmluciano** [comment]
I agree, I was rebasing on your work just to grab the CNI bits that you added.

**Comment by ozdanborne** [comment]
@cmluciano Now that the flannel PR has been merged, I'm hoping we can get this PR in.

I looked into the error and I believe it is caused by Calico choosing an [arbitrary service IP for calico-etcd](https://github.com/kubernetes/kubernetes-anywhere/pull/405/files#diff-1a1c0fa04d8afe13e2b967dc3505392eR4) that is outside [the service range configured by ignition](https://github.com/kubernetes/kubernetes-anywhere/blob/a27971e12fed1a0f78d965678e1f7c830a441f84/phase2/ignition/vanilla/manifest/kube-apiserver.jsonnet#L33). The fix should just be to update any reference to `10.96.232.136` in this PR (there should be at least 2) to, say, `10.0.232.136`.

Do you have bandwidth to get this PR rolling again?

**Comment by cmluciano** [comment]
Yes I can submit that patch. I've been meaning to test out https://github.com/kubernetes/kubernetes-anywhere/pull/432 with this.

**Review by jamiehannaford — COMMENTED** [review]
LGTM. Can you also update [this line](https://github.com/kubernetes/kubernetes-anywhere/blob/master/phase3/Kconfig#L51) to add Calico as a CNI option? (You might also want to correct the indentation while you're at it).

**Comment by cmluciano** [comment]
@jamiehannaford @caseydavenport Updated the CNI description and fixed the indent

**Comment by tmjd** [comment]
The PR on kubernetes/test-infra#4981 is ready to go when this PR gets merged. Is there anything blocking this currently (other than approvals)?

**Comment by jamiehannaford** [comment]
Just needs someone from @kubernetes/kubernetes-anywhere-reviewers for a lgtm

**Comment by pipejakob** [comment]
Taking a look now, but as a heads-up there's a merge conflict now that will need to be addressed.

**Comment by pipejakob** [comment]
Cool, I didn't meticulously go through all of the individual manifest files, so hopefully they match a canonical install recommendation (and are tested), but otherwise looks good.

/lgtm

I'll merge once you resolve conflicts.

**Comment by cmluciano** [comment]
@jamiehannaford @caseydavenport @tmjd 

Can you verify that the calico manifests look right to you?

**Comment by tmjd** [comment]
LGTM :+1:

**Review by caseydavenport — APPROVED** [review]
@cmluciano LGTM

**Pull request merged**